src/processor/operator/persistent/node_batch_insert.cpp

```diff
@@ -124,8 +128,21 @@ void NodeBatchInsert::executeInternal(ExecutionContext* context) {
         token = nodeLocalState->localIndexBuilder->getProducerToken();
     }
 
+    auto nodeInfo = ku_dynamic_cast<BatchInsertInfo*, NodeBatchInsertInfo*>(info.get());
     while (children[0]->getNextTuple(context)) {
         auto originalSelVector = nodeLocalState->columnState->getSelVectorShared();
+        auto numTuples = originalSelVector->getSelSize();
+        for (auto i = 0u; i < nodeInfo->columnPositions.size(); ++i) {
+            auto pos = nodeInfo->columnPositions[i];
+            auto defaultVector = nodeLocalState->columnVectors[i];
+            if (!pos.isValid()) {
```

Comment:
If we make the above change, you can no longer use invalid position to check for default expressions. You may need to pass additional field to indicate which expression is default and which is input.